### DISCUSSION TITLE
"debug test" not working anymore

### DISCUSSION THREAD

**Discussion by chappjc** [golang/vscode-go#2119] [Help] [answered]
After updating to Go 1.18 or maybe just vscode-go 0.32, clicking "debug test" for a unit test function no longer works in _any_ of my workspaces:

![image](https://user-images.githubusercontent.com/9373513/159023105-133fdf1f-dea0-4d66-8703-6e3b9df5f412.png)

extension v0.32.  All Go tools updated using the extension.

The launch.json varies between projects, but they are all fairly vanilla, and they all used to work fine.

e.g. 

```js
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Attach to Process",
            "type": "go",
            "request": "attach",
            "mode": "local",
            "processId": 3578
        }
    ]
}
```

Or even in a fresh workspace where it's just created automatically as `"configurations": []`.

Is this a bug or do I have something wrong on my end?  It suddenly started doing this for all my Go workspaces.

VSCode:

Version: 1.65.2
Commit: c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1
Date: 2022-03-10T19:23:12.185Z
Electron: 13.6.9
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.40-electron.0
OS: Linux x64 5.16.14-arch1-1

**Comment by hyangah** [comment, marked as answer]
As the error message says there seems to be an environment variable with numeric value.
Can you turn on the debug trace and share what `attach` request looks like?

**Reply by chappjc** [reply]
To be clear, I'm not using that example "Attach to Process" debug config.  I get the same results with `"configurations": []`.  I'm just clicking on "debug test" right above a unit test definition in the code.

I'm not sure where to "turn on the debug trace", but in the Debug Console, it just shows it starting dlv dap:

```
Starting: [/home/jon/go/bin/dlv]() dap --check-go-version=false --listen=127.0.0.1:33973 --log-dest=3 from [/home/jon/github/decred/dcrdex/server/auth]()
DAP server listening at: 127.0.0.1:33973
```

Then VS code throws that error.

**EDIT:** I just verified that I can still use the "Attach to Process" config and debug a running process, it's just the "debug test" functionality that's broken for me.

**Reply by chappjc** [reply]
I find the message `cannot unmarshal number into "env" of type string` hard to unpack, but it sounds like it's trying to make a debug config on-the-fly for the "debug test" click by writing something into an `"env"` field's key, but instead of writing a string into it, it's trying to write a number.

**Reply by chappjc** [reply]
Aha!  I had the following set in my settings.json:

```json
    "go.testEnvVars": {
        "CGO_ENABLED": 0
    },
```

had to change it to:

```json
    "go.testEnvVars": {
        "CGO_ENABLED": "0"
    },
```

Although, for future reference, I would like know how get the trace level vscode extension debugging needed to see the construction of this temporary debug config, as well as the location of the temporary debug config itself.
